src/RequestProcessor.cs

```diff
@@ -159,7 +159,10 @@ internal StreamingMessage ProcessFunctionLoadRequest(StreamingMessage request)
             // Ideally, the initialization should happen when processing 'WorkerInitRequest', however, the 'WorkerInitRequest'
             // message doesn't provide information about the FunctionApp. That information is not available until the first
             // 'FunctionLoadRequest' comes in. Therefore, we run initialization here.
-            if (!_isFunctionAppInitialized)
+            // Also, we receive a FunctionLoadRequest when a proxy is configured. Proxies don't have the Metadata.Directory set
+            // which would cause initialization issues with the PSModulePath. The only way to tell if a FunctionLoadRequest is
+            // from a proxy is to test if Metadata.Directory is null or empty which is what we do here.
+            if (!_isFunctionAppInitialized && !string.IsNullOrEmpty(functionLoadRequest.Metadata.Directory))
```

Comment:
Could you please use ```functionLoadRequest.Metadata.IsProxy```? I have updated the proto files for the worker, so this property is available now.

Reply:
Done!